easybuild/easyblocks/e/easybuildmeta.py

```diff
@@ -66,6 +66,28 @@ def __init__(self, *args, **kwargs):
             # consider setuptools first, in case it is listed as a sources
             self.easybuild_pkgs.insert(0, 'setuptools')
 
+        # opt-in to using pip for recent version of EasyBuild, if:
+        # - EasyBuild is being installed for Python >= 3.6;
+        # - pip is available, and recent enough (>= 21.0);
+        # - use_pip is not specified;
+        pyver = sys.version.split(' ')[0]
+        pip = which('pip')
```

Comment:
I think this is problematic, it doesn't take into account the potential usage of `EB_PYTHON`/`EB_INSTALLPYTHON` which means that `pip` may not even be in the users path but could still be usable.